### PR TITLE
Add WebUSB support to ashell

### PR DESCRIPTION
**Pull request by nagineni**
This adds WebUSB support to the ashell to allow the user to
upload JS code directly from the browser IDE to Zephyr.js
device for execution.

Fixes issue #606

Signed-off-by: Sudarsana Nagineni <sudarsana.nagineni@intel.com>

### CODE REVIEW

**Comment by grgustaf** [comment]
+1 LGTM

**Comment by poussa** [comment]
LGTM

**Comment by brianjjones** [comment]
LGTM

File: src/ashell/webusb-handler.c [hunk 1]
```diff
@@ -0,0 +1,185 @@
+// Copyright (c) 2016, Intel Corporation.
```
Comment (grgustaf) [inline src/ashell/webusb-handler.c]: Change to 2017

File: src/ashell/webusb-handler.h [hunk 2]
```diff
@@ -0,0 +1,10 @@
+// Copyright (c) 2016, Intel Corporation.
```
Comment (grgustaf) [inline src/ashell/webusb-handler.h]: And here too. :)

File: src/ashell/webusb-handler.c (hunk 1 above)
Comment (nagineni) [inline src/ashell/webusb-handler.c]: Fixed.

**Comment by nagineni** [comment]
@grgustaf @poussa @brianjjones Thank you all for the quick review!

**Pull request merged**